### PR TITLE
Break long lines without spaces - Fixes #12

### LINKED ISSUE #12: improve previous line count
The previous line count is being tracked by `log-update` but it does not take the terminal size into account.

For instance, if you log a line of 40 characters while the terminal is only 30 characters, it will be printed on 2 lines. After calling `log-update` again, it will only erase the 2nd line because the previous line count was only 1 and not 2.

So we have to iterate over all the lines [here](https://github.com/sindresorhus/log-update/blob/master/index.js#L12) and calculate the number of printed lines instead.

### PR DESCRIPTION
**Pull request by Jameskmonger**

### CODE REVIEW

**Comment by Jameskmonger** [comment]
FYI @SamVerschueren

**Comment by Jameskmonger** [comment]
Ignore this for now please @SamVerschueren (don't close) as I need to fix the output on Travis.

**Comment by Jameskmonger** [comment]
Now ready for you @SamVerschueren / @sindresorhus

**Comment by Jameskmonger** [comment]
Hey @SamVerschueren / @sindresorhus did you manage to take a look at this?

**Comment by OwenPattison** [comment]
This would be useful for me too if this could be looked at @sindresorhus @SamVerschueren

File: index.js [hunk 1]
```diff
@@ -6,10 +6,14 @@ const wrapAnsi = require('wrap-ansi');
 const main = stream => {
 	let prevLineCount = 0;
 
+	const getWidth = function (columns) {
+		return (columns && columns > 0) ? columns - 1 : 80;
```
Comment (kevva) [inline index.js]: Why this `(columns && columns > 0) ? columns - 1`?

File: index.js [hunk 2]
```diff
@@ -6,10 +6,14 @@ const wrapAnsi = require('wrap-ansi');
...
+	};
+
 	const render = function () {
 		cliCursor.hide();
 		let out = [].join.call(arguments, ' ') + '\n';
-		out = wrapAnsi(out, process.stdout.columns || 80, {wordWrap: false});
+		out = wrapAnsi(out, getWidth(process.stdout.columns), {hard: true, wordWrap: false});
```
Comment (kevva) [inline index.js]: Could you use https://github.com/sindresorhus/term-size instead of `process.stdout.columns` too?

Comment (SamVerschueren) [inline index.js]: Is that the reason why it fails on Windows? Because `process.stdout.columns` is not set correctly?

**Comment by SamVerschueren** [comment]
Can someone of you log `process.stdout.columns` on Windows? Is it set? Is it correct?

**Comment by sindresorhus** [comment]
Yup, it's set correctly:

screen shot 2017-07-23 at 14 04 55

File: index.js (hunk 2 above)
Comment (sindresorhus) [inline index.js]: I don't think term-size is needed here as this module wouldn't work when not TTY anyways.

**Comment by SamVerschueren** [comment]
So the only difference with what we had before is the `-1`...

**Comment by sindresorhus** [comment]
`{hard: true}` too.

**Comment by SamVerschueren** [comment]
I can see why `hard` should be set to `true`, but don't really get why `-1`.

**Comment by sindresorhus** [comment]
@SamVerschueren It's explained in #19, but I don't understand it either:

> Set `width` to `process.stdout.columns - 1` (otherwise there is a single blank line added)

**Comment by SamVerschueren** [comment]
Thanks for the pointer. I thought I read it somewhere but couldn't find it anymore. Apparently was reading in the wrong issue.

**Comment by sindresorhus** [comment]
I've done a new release now with just the `hard` option. Let's see what @Jameskmonger can elaborate on the `-1` thing.

**Comment by Jameskmonger** [comment]
Without `-1`, it doesn't work on Windows. I'm not sure why exactly, but if the console is 80 columns wide and you print 80 characters, it appears to print across two lines.

See the two screenshots below. The first is the most recent `log-update` version (without `-1`) and the second is the most recent `log-update` version but with a change to add `-1`.

Before change
After change

File: index.js (hunk 1 above)
Comment (Jameskmonger) [inline index.js]: `process.stdout.columns` isn't set on Travis CI.

**Comment by sindresorhus** [comment]
Alright, but then I think we should only do the `-1` thing on Windows. And add a short code comment on why it's there.

**Comment by Jameskmonger** [comment]
Sure. I'll get on that after work @sindresorhus

**Comment by sindresorhus** [comment]
Ping @Jameskmonger :)

**Comment by Jameskmonger** [comment]
Sorry @sindresorhus. Took me a long time to finish work 😅. I've pushed the changes (only `-1` on Windows) and also merged `master` to resolve the conflicts.

**Comment by OwenPattison** [comment]
@sindresorhus Please could we get this merged asap as we have a dependency upon this :(, thanks.

**Comment by Jameskmonger** [comment]
Thank you @sindresorhus :heart:

**Pull request merged**